Zmdernar/top stories select

Zachary-Dernar opened the pull request.

Zachary-Dernar committed:
Adds sorting by date to top stories component

Still needs design work

Zachary-Dernar committed:
Merge branch 'master' of https://github.com/watson-developer-cloud/discovery-nodejs into zmdernar/top-stories-select

Zachary-Dernar committed:
Merge branch 'master' of https://github.com/watson-developer-cloud/discovery-nodejs into zmdernar/top-stories-select

Zachary-Dernar committed:
garbage commit

Zachary-Dernar committed:
garbage commit 2

Zachary-Dernar committed:
garbage commit 3

Zachary-Dernar committed:
Fixes the style.css file

Zachary-Dernar committed:
Merge branch 'master' into zmdernar/top-stories-select

Zachary-Dernar committed:
Updates design for sort select

Changes icon size and moves score

maniax89 commented on views/TopStories/index.jsx:
```diff
@@ -50,6 +80,15 @@ export default React.createClass({
 
+  onChangeSort(e) {
+    this.setState({ sortType: e.target.value });
+    let sortVal = e.target.value == 'date' ? '-blekko.chrondate,-_score' : '';
```
now that I think about it - would it be better to pass 'date' or 'relevance' to the `query.sort` value? then in the `query-builder.js` you do this same test. will keep the notion of fields in the `query-builder.js` instead of scattered in the `TopStories` component

maniax89 commented on views/TopStories/index.jsx:
```diff
@@ -13,12 +18,35 @@ const Story = props => (
+  </span>
+);
+
+//<Icon type="right" />
```
remove this

maniax89 commented on views/TopStories/index.jsx:
```diff
@@ -13,12 +18,35 @@ const Story = props => (
+
+//<Icon type="right" />
+
+//<option value="relevance" {this.state.sortType == 'Relevance' ? 'selected' : ''}>Relevance</option>
```
remove this

maniax89 reviewed (commented):
can we come up with a test for this?

Zachary-Dernar committed:
Moves sort query value management to query-builder.js

This allows for a more readable TopStories/index.js

Zachary-Dernar committed:
Adds test for TopStories sort functionality

Introduces sinon into dependencies

Zachary-Dernar committed:
Cleans up test for Top Stories Sort Select

Reformats tests, including pulling the selecting functionality
into separate functions

maniax89 commented on test/views/TopStories/test.index.js:
```diff
@@ -0,0 +1,50 @@
+    });
+  });
+
+  describe('When I select Relevance', () => {
```
i would put this inside the 'When I select date' `describe` (same level as `it`) and phrase it like 'then I select relevance'

maniax89 commented on test/views/TopStories/test.index.js:
```diff
@@ -0,0 +1,50 @@
+    });
+
+    it('calls onSortChange with Relevance', () => {
+      assert.ok(onSortChangeSpy.calledWith(expected));
```
i am concerned that this passes. `expected` has `date` as the `sort`, so seeing that it is called with that isn't testing what this test says it is testing. you probably should have it check both then if you want to validate both sort change events happen

maniax89 commented on test/views/TopStories/test.index.js:
```diff
@@ -0,0 +1,50 @@
+  const expected = Object.assign({}, query_sample, {
+    sort: 'date'
+  });
+  let wrapper = shallow(<TopStories stories={stories} query={query_sample} onSortChange={onSortChangeSpy} />);
```
this should be inside the `beforeEach`

maniax89 commented on test/views/TopStories/test.index.js:
```diff
@@ -0,0 +1,50 @@
+  storiesWrapper.find('.sort-option').nodes[0].props.children[1].props.children.props.onChange({target: { value : 'date'}});
+}
+
+function selectRelevance(storiesWrapper) {
```
you could just have 1 method called `selectOption` and pass in the wrapper + the value you want to select -> `selectOption(wrapper, 'date')`

maniax89 replied:
it would be even better if you didn't have to pass in the wrapper at all and just refer to it within the describe blocks - javascript functions are hoisted so you can just embed them within the describe block and they will have access to your `let wrapper;` that ultimately gets assigned within the `beforeEach`

maniax89 commented on test/views/TopStories/test.index.js:
```diff
@@ -0,0 +1,50 @@
+    'text': 'Sample Company'
+  };
+  const onSortChangeSpy = sinon.spy();
+  const expected = Object.assign({}, query_sample, {
```
this should be inside the `it` for each test

Zachary-Dernar committed:
Merge branch 'master' into zmdernar/top-stories-select

Zachary-Dernar committed:
Improves test architecture

Better describes the scope and changes for each variable and properly
tests each result

Zachary-Dernar committed:
Merge branch 'master' of https://github.com/watson-developer-cloud/discovery-nodejs into zmdernar/top-stories-select

Zachary-Dernar committed:
Merge branch 'zmdernar/top-stories-select' of https://github.com/watson-developer-cloud/discovery-nodejs into zmdernar/top-stories-select
